diff --git a/backends/imgui_impl_opengl3.h b/backends/imgui_impl_opengl3.h
index b72b5c887d63..deff0b298413 100644
--- a/backends/imgui_impl_opengl3.h
+++ b/backends/imgui_impl_opengl3.h
@@ -47,6 +47,10 @@ IMGUI_IMPL_API void     ImGui_ImplOpenGL3_UpdateTexture(ImTextureData* tex);
 //#define IMGUI_IMPL_OPENGL_ES2     // Enable ES 2 (Auto-detected on Emscripten)
 //#define IMGUI_IMPL_OPENGL_ES3     // Enable ES 3 (Auto-detected on iOS/Android)
 
+#ifndef IMGUI_IMPL_OPENGL_ES3
+#define IMGUI_IMPL_OPENGL_ES3
+#endif
+
 // You can explicitly select GLES2 or GLES3 API by using one of the '#define IMGUI_IMPL_OPENGL_LOADER_XXX' in imconfig.h or compiler command-line.
 #if !defined(IMGUI_IMPL_OPENGL_ES2) \
  && !defined(IMGUI_IMPL_OPENGL_ES3)
